### ISSUE TITLE
Bug: `no-extra-bind` fails to flag unnecessary `.bind()` when `this` is used inside class fields or static blocks

### ISSUE THREAD

**Issue by synthex-byte** [eslint/eslint#21258]
### Environment

ESLint Playground

### What parser are you using?

Default (Espree)

### Playground Code
```js
const f = (function() {
    class C {
        field = this.x;
    }
}).bind(this);
```

### ESLint Configuration 
<details> 
 <summary> Tap to see full config </summary> 

```js
import { defineConfig } from "eslint/config";

export default defineConfig([
    {
        "rules": {
            "no-extra-bind": [
                "error"
            ]
        },
        "languageOptions": {
            "ecmaVersion": "latest",
            "sourceType": "module"
        }
    }
]);
```
 </details>

### What did you expect to happen?

ESLint should report `"The function binding is unnecessary."`.
The outer functions don't use `this` in their own scope — the `this` inside class fields and static blocks belongs to the class itself (instance for fields, constructor for static blocks), not the enclosing function. So `.bind(this)` on the outer function is redundant.

### What actually happened?

No problem is reported.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgZiA9IChmdW5jdGlvbigpIHtcbiAgICBjbGFzcyBDIHtcbiAgICAgICAgZmllbGQgPSB0aGlzLng7XG4gICAgfVxufSkuYmluZCh0aGlzKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLWV4dHJhLWJpbmQiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

The same issue occurs with static blocks

**Comment by mdjermanovic** [comment]
Thanks for the issue. I agree this is a bug, a PR is welcome.
